sdk/storage/azblob/container/responses.go

```diff
@@ -42,3 +43,12 @@ type GetAccessPolicyResponse = generated.ContainerClientGetAccessPolicyResponse
 
 // SetAccessPolicyResponse contains the response from method Client.SetAccessPolicy.
 type SetAccessPolicyResponse = generated.ContainerClientSetAccessPolicyResponse
+
+// SubmitBatchResponse contains the response from method Client.SubmitBatch.
+type SubmitBatchResponse struct {
+	generated.ContainerClientSubmitBatchResponse
```

Comment:
We typically don't embed unexported types in public surface area.  Can't the generated type be used and exported?

Reply:
The response body that comes from the generated code is like [this](https://learn.microsoft.com/en-us/rest/api/storageservices/blob-batch#sample-response).
Here I am parsing this response body and storing the responses of the individual sub-requests. The reason why I have kept `generated.ContainerClientSubmitBatchResponse` is because if the parsing of the sub-request's response results in error due to unknown reasons, the users can always refer the raw response from the `generated.ContainerClientSubmitBatchResponse`.

If we want to export the generated response, we can do something like this.
```
type BatchRawResponse = generated.ContainerClientSubmitBatchResponse

// SubmitBatchResponse contains the response from method Client.SubmitBatch.
type SubmitBatchResponse struct {
	BatchRawResponse
	Responses []*BatchSubResponse
}
```

Reply:
If the parsing fails, I would expect a zero-value `SubmitBatchResponse` to be returned a long with the appropriate `error`.

Reply:
Here we are talking about parsing of the individual responses for the sub-requests. If parsing fails for one of the sub-request's response, then we should not fail the entire batch IMO. For these cases, we will store the error in the individual response object(`BatchSubResponse`) in the `Error` field. Users can always refer the raw response stored in the `BatchRawResponse` field of the `SubmitBatchResponse`.
```
// BatchSubResponse contains the response for the individual sub-requests.
type BatchSubResponse struct {
	ContentID     *int
	ContainerName *string
	BlobName      *string
	Response      *http.Response
	Error         error
}
```

Reply:
In Go, when a func/method returns `(Foo, error)`, and the `error` value is not `nil`, callers must _not_ inspect the value of `Foo` as it's unspecified.  It's best to make this the zero-value so we don't inadvertently let callers violate this principle.

If `SubmitBatch()` fails to parse the response, we want to directly return the parsing error so it's clear we have a bug that needs to be fixed (we do the same thing for all of the other APIs).

Regarding the raw response, we've removed this from the success case.  If callers need access to the raw HTTP response, they must use [`runtime.WithCaptureResponse`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.3.1/runtime#WithCaptureResponse) to get it.